src/yamerl_node_map.erl

```diff
@@ -44,6 +44,11 @@
 
 -define(TAG, "tag:yaml.org,2002:map").
 
+-record(map_builder,
+    {format,
+     state,
+     keys,
```

Comment:
What do you think of using a map instead of a proplist to keep track of existing keys?

Reply:
Using a map would definitely simplify this code I think. I was unsure what versions of Erlang this library was trying to support so I kept it as a proplist. For my needs, I just need this to be compatible with OTP 17 (which introduced maps).